site/en/docs/web-platform/popup-api/index.md

```diff
@@ -0,0 +1,209 @@
+
+[`popUpShowTarget`](/docs/web-platform/popup-api/popupshowtarget-property)
+: Creates a trigger element that shows an associated hidden pop-up. DOM definition of the HTML [`popupshowtarget`](/docs/web-platform/popup-api/popupshowtarget-attribute) attribute.
+
+[`popUpHideTarget`](/docs/web-platform/popup-api/popuphidetarget-property)
+: Creates a trigger element that dismisses an associated shown pop-up. DOM definition of the HTML [`popuphidetarget`](/docs/web-platform/popup-api/popuphidetarget-attribute) attribute.
+
+## Examples
+
+{% Aside %}
+This section shows basic usage of the API; for more substantial code including numerous complete examples, check out [Pop-ups: They're making a resurgence!](/blog/pop-ups-theyre-making-a-resurgence/) by Jhey Tompkins.
+{% endAside %}
+
+Declaratively create different types of pop-up. The first two are equivalent:
+
+```html
+<p id="my-popup" popup>Hello!</p>
+<p id="my-popup" popup="auto">Hello!</p>
+<p id="my-popup" popup="hint">Hello!</p>
+<p id="my-popup" popup="manual">Hello!</p>
+```
+
+Specify that a pop-up should open automatically on page load with [`defaultopen`](/docs/web-platform/popup-api/defaultopen-attribute):
+
+```html
+<p id="my-popup" popup="auto" defaultopen>Hello!</p>
+```
+
+{% Aside 'caution' %}
+[`defaultopen`](/docs/web-platform/popup-api/defaultopen-attribute) does not work with pop-ups of type `hint`.
+{% endAside %}
+
+Declaratively create buttons to show and hide the pop-up:
+
+```html
+<button popupshowtarget="my-popup">Show pop-up</button>
+<button popuphidetarget="my-popup">Dismiss pop-up</button>
+```
+
+Declaratively create a single button to toggle the pop-up between shown and hidden states:
+
+```html
+<button popuptoggletarget="my-popup">Show pop-up</button>
+```
+
+Use the [`:open`](/docs/web-platform/popup-api/open-pseudo-class) pseudo-class to style the pop-up when it has been shown. In this case we are transitioning the pop-up in from the bottom of the viewport, rather than just having it appear in the center:
+
+```css
+p[popup] {
+  transform: translateY(60vh);
+}
+
+p[popup]:open {
+  transition: transform 0.6s;
+  transform: translateY(0vh);
+}
+```
+
+Use the [`::backdrop`](/docs/web-platform/popup-api/backdrop-pseudo-element) pseudo-element to style the document behind the pop-up while it is being shown — here we are darkening the rest of the content:
+
+```css
+p[popup]::backdrop {
+  background-color: rgba(0,0,0,0.5);
+}
+```
+
+Use JavaScript to programmatically show and hide pop-ups:
+
+```js
+  popup.showPopUp();
+  popup.hidePopUp();
+```
+
+Use event listeners to respond to pop-ups being shown and hidden. Here we are switching the `textContent` of the pop-up trigger button so that it makes sense when the pop-up is in its shown and hidden states:
```

Comment:
```suggestion
Use event listeners to respond to pop-ups being shown and hidden. The following example switches the `textContent` of the pop-up trigger button so that it makes sense when the pop-up is in its shown and hidden states:
```